core/modulesource.go

```diff
@@ -245,6 +249,150 @@ func (src *ModuleSource) AutomaticGitignore(ctx context.Context) (*bool, error)
 	return modCfg.Codegen.AutomaticGitignore, nil
 }
 
+// LoadContext loads a directory from the module context directory.
+//
+// If the module is local, it will load the directory from the local source
+// directly from the host.
+//
+// If the module is git, it will load the directory from the git repository
+// using its context directory.
+//
+// The path is relative to the module context directory, even if it's absolute.
+//
+// Improvement for later: use `.gitignore` to excludes paths from the context.
+func (src *ModuleSource) LoadContext(ctx context.Context, dag *dagql.Server, path string, ignore []string) (inst dagql.Instance[*Directory], err error) {
+	// We exclude .git by default because it's not useful and tends to invalidate the cache.
+	excludes := []string{"**/.git"}
+	includes := []string{}
```

Comment:
I don't think we should apply this by default - there are legitimate cases for needing the `.git` directory, like getting commit ids, etc. There's no way to disable this really, and it's a bit non-obvious.

Ideally, we'd be able to use context directories *in combination* with some `include`/`exclude`s/views in `dagger.json`, so `.git` could be disabled globally there - but I don't think we should apply it as a default for everywhere.

Reply:
I believe that was copied from here:

https://github.com/dagger/dagger/blob/540d884cf4583313356ebcb21b97da2b926122d1/core/schema/modulesource.go#L758-L761

I think it's fine, if the function author wants to allow `.git` they can add `!**/.git` to `// +ignore`.

Reply:
Also FTR, we're tying to see if this feature can replace _views_, so consider it might get deprecated.

Reply:
I agree that we should not exclude `.git` by default in this exact place.

The place where `.git` got excluded previously was specifically when loading the module source that was given to SDKs for building the module. For SDKs, they don't (currently) care about `.git` so it was indeed a waste of time and invalidated cache more often than necessary. For that case specifically, it's probably worth keeping the default exclude of `.git`.

But in the general case of loading any contextual directory, I don't think it should be excluded by default. Like @jedevc said there are legit use cases for it.